### DISCUSSION TITLE
Canvas usage with Barryvdh\DomPDF

### DISCUSSION THREAD

**Discussion by andsvi** [dompdf/dompdf#3551] [General]
I was in Barryvdh\DomPDF discussions and every discussion about adding pages to each pdf page footer led me here. So if I want to implement pages number in each pdf page I cant use Barryvdh\DomPDF and must use this?

**Comment by bsweeney** [comment]
Without more context I can't say what you need. Barryvdh is just a wrapper around Dompdf, so as far as I know you do not need to switch to straight Dompdf usage.

Page numbers are supported through CSS, but if you want to also display the total number of pages then you'll need to do it through scripting.

**Reply by andsvi** [reply]
Right now how I use Barryvdh version and never even make Dompdf object, I am always working with Barryvdh\DomPDF\Facade\Pdf object. I give blade template to it with 

$pdf = Pdf::loadView('files.pdf-template', $data);
 return $pdf->stream('announcement.pdf');
 
 I tried to extract canvas from pdf element itself but I think it can only be done from Dompdf object

**Reply by bsweeney** [reply]
The [`getDomPDF` method](https://github.com/barryvdh/laravel-dompdf/blob/v3.0.0/src/PDF.php#L82-L88) allows you to retrive the Dompdf instance.
